BootloaderCorePkg/Stage2/Stage2.c

```diff
@@ -254,19 +251,12 @@ NormalBootPath (
   BoardInit (EndOfStages);
 
   PayloadId = GetPayloadId ();
-  if (PayloadId == 0) {
-    // For built-in payload including OsLoader and FirmwareUpdate, it will handle
-    // notification through SBL platform services, so do not call notifications
-    // here.
-    CallBoardNotify = FALSE;
-  } else if ((PayloadId == UEFI_PAYLOAD_ID_SIGNATURE) && (UefiSig != 0)) {
-    // Current open sourced UEFI payload does not call any FSP notifications,
-    // but some customized UEFI payload will. The 1st DWORD in UEFI payload image
-    // will be used to indicate if it will handle FSP notifications.
-    CallBoardNotify = FALSE;
-  } else {
-    CallBoardNotify = TRUE;
-  }
+
+  // Force board notify to false here to avoid going
+  // into a low power state that prevents SMM
+  // initialization in MTL and RPL universal payloads
+  CallBoardNotify = FALSE;
```

Comment:
i believe we should have a condition for other payloads apart from osloader and UEFIpayload.